ivi_water/api_client.py

```
@@ -180,8 +187,12 @@ def _check_redirect_security(self, response: requests.Response, *args, **kwargs)
                     self._validate_base_url(full_url)
                 except ValueError as e:
                     safe_log_url = sanitize_for_terminal(redact_url(full_url))
-                    self.logger.critical(f"Security check failed for redirect to {safe_log_url}: {e}")
-                    raise requests.exceptions.ConnectionError(f"Security check failed for redirect: {e}")
+                    self.logger.critical(
+                        f"Security check failed for redirect to {safe_log_url}: {e}"
+                    )
+                    raise requests.exceptions.ConnectionError(
+                        f"Security check failed for redirect: {e}"
+                    )
```

Ruff (0.15.6)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)